### PR TITLE
[USER]: add tests for service layer

### PR DESCRIPTION
**Pull request by IhorKalaur**

### CODE REVIEW

File: src/main/java/com/team01/carsharingapp/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -42,32 +41,27 @@ public UserRegistrationResponseDto register(UserRegistrationRequestDto requestDt
...
 
         return userMapper.toDto(userRepository.save(user));
     }
 
     @Override
     @Transactional
-    public UserDto update(UpdateUserDto updateUserDto) {
-        User userFromDB = getUserById(getCurrentUser().getId());
-        userRepository.findById(getCurrentUser().getId());
+    public UserDto update(UpdateUserDto updateUserDto, User user) {
```
Comment (Someboty) [inline src/main/java/com/team01/carsharingapp/service/impl/UserServiceImpl.java:63]: Where do you use user in your method?

**Review by Someboty — APPROVED** [review]
GJ! Can you add coverage of your tests?

File: src/test/java/com/team01/carsharingapp/service/impl/UserServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+        when(userMapper.toResponseDto(user))
+                .thenReturn(getUserRegistrationResponseDtoFromUser(user));
+
+        UserRegistrationResponseDto expected = getUserRegistrationResponseDto();
+        UserRegistrationResponseDto actual = userServiceImpl.register(requestDto);
+
+        assertNotNull(actual);
+        assertEquals(expected, actual);
+        verify(userRepository, times(1)).save(user);
+        verify(userMapper, times(1)).toResponseDto(user);
```
Comment (Denis-Balako) [inline src/test/java/com/team01/carsharingapp/service/impl/UserServiceImplTest.java:83]: times(1) already set by default. You don't need to verify methods, if they use in when() method

**Review by Denis-Balako — CHANGES_REQUESTED** [review]
Good job! Left some recommendations

**Review by Denis-Balako — APPROVED** [review]
LGTM!

**Pull request merged**